### DISCUSSION TITLE
Documentation Update - OIDC Logout URI Requires both OIDC_LOGOUT_URI and OIDC_DISABLE_REDIRECT

### DISCUSSION THREAD

**Discussion by cryptelli** [outline/outline#7382] [General]
Hi all,

I've been really enjoying Outline however was having issues with it logging back in each time I logged out using Authentik. I was thrown off by the below documentation available from [here](https://docs.getoutline.com/s/hosting/doc/oidc-8CPBm6uC0I), which states that you need one or the other.

However, based on my experience you need both configured. I can now logout, be redirected to Authentik as expected.

`OIDC_DISABLE_REDIRECT` – If set, will prevent Outline from automatically redirecting to the OIDC provider
`OIDC_LOGOUT_URI` – A url that the client will be redirect to post-logout

![image](https://github.com/user-attachments/assets/96f4c24e-8c7d-45de-99c7-a2cf2b0baba7)

I hope this helps someone else in a similar situation.

**Comment by tommoor** [comment]
I believe the documentation makes sense, you _either_ need to choose a location to redirect to post-logout or disable the automatic redirect to the login endpoint.

The problem may be that your `OIDC_LOGOUT_URI` is redirecting **back** to the Outline login page? In which case, yes, you would need both.

**Reply by cryptelli** [reply]
I am using the logout URI which looks like https://x.domain.name/application/o/outline/end-session/. Without configuring both it was in an endless loop.

So not sure if this is an Authentik thing or Outline, have other services configured using the same OIDC URI and they work fine.

**Reply by tommoor** [reply]
I mean, where do you get sent to after hitting that url? – it should be easy to determine.

**Reply by cryptelli** [reply]
It hits https://x.domain.name/if/flow/default-provider-authorization-explicit-consent/?response_type=code&redirect_uri=https%3A%2F%2Foutline.domain.name%2Fauth%2Foidc.callback.

**Comment by ovizii** [comment]
I think I am having the exact same problem. I am using Pocket ID as the IDP.

Here is my flow:

`visit wiki.dom.tld => get redirected to IDP login page => Login => wiki.dom.tld => click Log out => end up at Log Out Page of IDP => click Sign out => end up on Login Page of IDP => click Authenticate => logged into auth.dom.tld instead of going back to wiki.dom.tld`

this variable is set:
`OIDC_LOGOUT_URI=https://auth.dom.tld/api/oidc/end-session`

tha description says:

> OIDC_LOGOUT_URI – A url that the client will be redirect to post-logout

But does outline provide post_logout_redirect_uri and id_token_hint to my IDP? 

Also, my IDP has a field called Logout Callback URL with the description:

> URL(s) provided by your client for logout. Wildcards (*) are supported, but best avoided for better security.

**What would I enter here?**
